src/pg/result_set.cr

```diff
@@ -15,7 +15,7 @@ class PG::ResultSet < ::DB::ResultSet
     @column_index = -1 # The current column
     @end = false       # Did we read all the rows?
     @rows_affected = 0_i64
-    @sized_io = Buffer.new(conn.soc, 1, statement.connection)
+    @sized_io = Buffer.new(conn.soc, 1, statement.connection.as(PG::Connection))
```

Comment:
There is a `conn` method that does this and is even used here in `conn.soc`. Doesn't this work?

```suggestion
    @sized_io = Buffer.new(conn.soc, 1, conn)
```

Reply:
Almost, but nope:

```
Error: expected argument #3 to 'PG::ResultSet::Buffer.new' to be PG::Connection, not PQ::Connection
```

I tried to make Buffer take a PQ::Connection, but it breaks further down the line:

```
In lib/pg/src/pg/decoder.cr:428:33

 428 | time.in io.connection.time_zone
                             ^--------
Error: undefined method 'time_zone' for PQ::Connection
```

Last attempt, to no avail:

```
 18 | @sized_io = Buffer.new(conn.soc, 1, conn.as(PG::Connection))
                                          ^
Error: can't cast PQ::Connection to PG::Connection
```

Reply:
Oh, PG vs PQ.